### ISSUE TITLE
Remote images

### ISSUE THREAD

**Issue by brh55** [brh55/react-native-masonry#16]
👎  Looks like I may have missed it during development of v0.1.0.

Feature:
- Allow remote images to be passed through

Validate:
- Pass in a remote image, it should load in the columns

**Comment by dazziola** [comment]
Does this mean the example is void? I had a look through your code and it appears that you are indeed using uri in the source prop of the Image component:

        <Image
            key={image.uri}
            source={{ uri: image.uri }}
            resizeMethod='auto'
            style={{ width: image.width, height: image.height, marginTop: gutter }} />

Just trying to get it running myself with remote images!

**Comment by brh55** [comment]
@dazziola 
Ahh my apologies, I meant to put "local" images. But yes, remote images will work. Let me know if you are having issues with getting it to work.

**Comment by dazziola** [comment]
Hmmm, I'm having a weird non-descript problem. I'm using the data block of images as per your example with two columns, however they're not rendering in the simulator or on a physical device.

**Comment by brh55** [comment]
@dazziola What version of RN and React are you running?

**Comment by dazziola** [comment]
"react-native": "0.43.3",
"react": "16.0.0-alpha.6",

**Comment by brh55** [comment]
@dazziola 
Hmm I just ran a fresh install of the example, no issues currently. Are your images HTTPS or HTTP?  If you are on IOS, by default IOS causes non-secure images to fail (https://github.com/facebook/react-native/issues/8520).

**Comment by dazziola** [comment]
I've cloned the example and you're right, it works perfectly. Hmmm, I'll mess around with my one and see if I can figure out what it is. Thanks for the prompt responses! If it's anything other than my own silly oversight, I'll be sure to post it here for future people!

**Comment by brh55** [comment]
@dazziola 
Absolutely! Please let me know what the issue was, maybe we can clear up or improve the readme 👍 .

**Comment by brh55** [comment]
Still currently trying to resolve this issue without the need to require a local image prop or something like that. Ideally I want the users to be able to set `{ uri: localFilePath }`, which currently resolves to a number -- I believe is the id within the asset manager. If anyone knows, please lmk!

**Comment by fxfactorial** [comment]
@brh55 Well why are you assuming    `  source={{ uri: image.uri }}`, I wouldn't do that. Just let the user pass whatever, you're just a front for the Image API

**Comment by brh55** [comment]
@fxfactorial Would work, but then local images wouldn't be able to leverage other properties such as `onPress, renderHeader, renderFooter`.

**Comment by fxfactorial** [comment]
@brh55 How so?

**Comment by brh55** [comment]
@fxfactorial  I'm assuming you are referring to users passing in react elements like 
`bricks = [<Image source={require('../asset1.jpg')} />, <Image source={uri: 'http://image.com/url1.jpg' } />]`?

**Comment by fxfactorial** [comment]
Hmm, I wasn't but that seems like a fine solution as well.

**Comment by brh55** [comment]
Ahh what other way are you referring to?

**Comment by fxfactorial** [comment]
`bricks = [{img: require('../asset1.jpg')}, {img: {uri: 'https:....'}}]`

I mean, just a front for the source prop of Image.

**Comment by brh55** [comment]
OHH yeah that should work, I'm not sure why I was having issues before.

**Comment by fxfactorial** [comment]
@brh55  actually now I'm thinking, should end user provide their own Image component, or whatever view they give. For example, what if I wanted to use this: https://github.com/oblador/react-native-image-progress

**Comment by fxfactorial** [comment]
How do you mean to overload?

**Comment by brh55** [comment]
Ah let me rephrase that -- I think it would be neat if you could use a custom image component, and it's probably possible.

By having a property that could allow users to pass a component for RNM to use in rendering process instead of the default image component  -- assuming the third-party component had `source` as a property.

**Comment by fxfactorial** [comment]
right, definitely some room for generalization

**Comment by brh55** [comment]
Indeed, theoretically it should work.

**Comment by brh55** [comment]
Per suggestions, there is still some underlying bundling that is occurring not allowing it to work as imagined technically. Going to leave this open, and welcomed for anyone to try and come up with a viable solution.
